build.rs

```diff
@@ -1,7 +1,7 @@
-extern crate winres;
+extern crate winresource;
 
 fn main() {
-    if cfg!(target_os = "windows") {
-        winres::WindowsResource::new().compile().unwrap();
+    if std::env::var("CARGO_CFG_TARGET_OS").unwrap() == "windows" {
```

Comment:
What's the behavior difference here?

Reply:
To quote winresource documentation:

> Please note: Using #[cfg(target_os = "windows")] in build.rs may not work as expected because build.rs is executed on the host. This means that target_os is always equal to host_os when compiling build.rs. E.g. if we use rustc on Linux and want to cross-compile binaries that run on Windows, target_os in build.rs is "linux". However, CARGO_CFG_TARGET_OS should always be defined and contains the actual target operating system, though it can only be checked at runtime of the build script.

Reply:
Ah, so that's because build script is always compiled for the host os. Understood, what an easy footgun 😅